### ISSUE TITLE
`NetworkedVector` rehaul

### ISSUE THREAD

**Issue by KillStr3aK** [roflmuffin/CounterStrikeSharp#135]
As far as I know `CUtlVectorEmbeddedNetworkVar<>` (`NetworkedVector<>`) is not supposed to only work with `CHandle<>` instances, and they can be used/treated like `CUtlVector<>` 

is this a typo? [CUtlVector\<CEntityHandle\>](https://github.com/roflmuffin/CounterStrikeSharp/blob/main/src/scripting/natives/natives_memory.cpp#L156)

also there is no point in implementing `IReadOnlyCollection<>` as we should be able to add new elements to the vector.

**Comment by roflmuffin** [comment]
Yeah this requires a full on managed side write I imagine, I did just entity handles on C++ side since the implementation exists in hl2sdk but I think it requires knowledge of the size of the underlying type?

**Comment by KillStr3aK** [comment]
When this will be worked on, lets not forget about #613

**Comment by KillStr3aK** [comment]
I believe if that pr is shipped with that `MemAlloc` class we could have an implementation for these ˙CUtl*˙ classes
